client/verta/verta/_internal_utils/_utils.py

```diff
@@ -259,19 +260,55 @@ def make_request(method, url, conn, **kwargs):
     with requests.Session() as s:
         s.mount(url, HTTPAdapter(max_retries=conn.retry))
         try:
-            response = s.request(method, url, **kwargs)
+            request = requests.Request(method, url, **kwargs).prepare()
+            response = s.send(request, allow_redirects=False)
+
+            # store initial response to add back into final history
+            initial_response = response
+
+            # manually inspect redirects to stop on 302s
+            responses = itertools.chain([initial_response], s.resolve_redirects(initial_response, request))
+            for response in responses:
+                if response.status_code == 302:
+                    if not conn.ignore_conn_err:
+                        raise RuntimeError(
+                            "received status 302 from {},"
+                            " which is not supported by the Client".format(response.url)
+                        )
+                    else:
+                        return fabricate_200()
+
+            if response is not initial_response:
+                # insert initial response back into history, b/c `resolve_redirects()` removed it
+                response.history.insert(0, initial_response)
```

Comment:
I don't understand why this is necessary or what it is doing exactly

Reply:
`requests.Response` has a `history` sequence attribute that keeps track of redirects that occurred.
Because of how `requests` implemented `resolve_redirects()`, the initial response isn't included in the beginning of that history.

This is only necessary if something downstream needs an accurate response history, which my tests do.

Reply:
Oh, interesting. I'd assume history is irrelevant because it's not following redirects. Can you add a comment clarifying this?

Reply:
Oh ugh you're right, thanks. I failed to catch this but there actually isn't _any_ history recorded in the final response.

I'd still prefer to have that full history intact though, both for testing and because I think the caller shouldn't lose access to it just because redirects are being done manually.